<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="description" content="Instructions for OLC Redmine Usage" /><link rel="canonical" href="https://olc-bioinformatics.github.io/redmine-docs/" />
      <link rel="shortcut icon" href="img/favicon.ico" />
    <title>OLC Redmine Automator</title>
    <link rel="stylesheet" href="css/theme.css" />
    <link rel="stylesheet" href="css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "Home";
        var mkdocs_page_input_path = "index.md";
        var mkdocs_page_url = "/redmine-docs/";
      </script>
    
    <!--[if lt IE 9]>
      <script src="js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
      <script>hljs.highlightAll();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href="." class="icon icon-home"> OLC Redmine Automator
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="./search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
                <li class="toctree-l1 current"><a class="reference internal current" href="#">Home</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#possible-analyses">Possible Analyses</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#i-want-to">I want to:</a>
    </li>
        </ul>
    </li>
    </ul>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="getting_started/">Getting Started</a>
                </li>
              </ul>
              <p class="caption"><span class="caption-text">Data</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="data/external_retrieve/">External Retrieve</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="data/report_retrieve/">Report Retrieve</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="data/sra_upload/">SRA Upload</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="data/sra_download/">SRA Download</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Analysis</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/amrsummary/">AMRsummary</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/autoclark/">AutoCLARK</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/bcgtree/">bcgTree</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/cardrgi/">CARD-RGI (AMR)</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/confindr/">ConFindr</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/cowsnphr/">COWSNPhR</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/closerelatives/">CloseRelatives</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/eCGF/">eCGF</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/ectyper/">ECTyper</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/diversitree/">DiversiTree</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/drep/">dRep-ANI</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/fastqc/">Fastqc</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/geneseekr/">GeneSeekr</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/gfa_retrieve/">GFA Retrieve</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/pyseer/">GWAS-pyseer</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/intimintyper/">IntiminTyper</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/kma/">KMA - Gene Detection</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/kraken2/">Kraken2</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/mashtree/">Mashtree</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/metaphlan/">Metaphlan4</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/mobsuite/">MobSuite</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/mlst/">MLST</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/neartree/">NearTree</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/plasmidborneidentity/">PlasmidBorneIdentity</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/pointfinder/">PointFinder</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/primerfinder/">PrimerFinder</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/prokka/">Prokka</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/psortb/">PSORTb</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/resfinder/">ResFinder</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/roary/">Roary</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/sequence_extractor/">SequenceExtractor</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/sipprverse/">sipprverse</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/snvphyl/">SNVPhyl</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/snippy/">Snippy</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/staramr/">staramr</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/strainmash/">StrainMash</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/unknownisolate/">Unknownisolate</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="analysis/wgsassembly/">WGS Assembly</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Internal Use</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="internal_only/merge/">Merge</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="internal_only/azuredownload/">AzureDownload</a>
                  </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="troubleshooting/">Troubleshooting</a>
                </li>
              </ul>
              <p class="caption"><span class="caption-text">Tutorials</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="tutorials/open_environment/">Open Environment</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="tutorials/create_pages/">Create Pages</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="tutorials/edit_pages/">Edit Pages</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="tutorials/publish_changes/">Publish Changes</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href=".">OLC Redmine Automator</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="." class="icon icon-home" aria-label="Docs"></a></li>
      <li class="breadcrumb-item active">Home</li>
    <li class="wy-breadcrumbs-aside">
          <a href="https://github.com/OLC-LOC-Bioinformatics/OLCRedmineAutomator/edit/master/docs/index.md" class="icon icon-github"> Edit on GitHub</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h1 id="olc-redmine-automator">OLC Redmine Automator</h1>
<p>To facilitate the sharing of the CFIA's genomic data and tools to analyze this data, we make use of the Redmine
platform. Our instance of Redmine can be found <a href="https://redmine.biodiversity.agr.gc.ca">here</a> (note that you must be on
the CFIA network in order to access Redmine).</p>
<p>If you have not yet gotten started with Redmine or want to know how the Redmine automator works and what it can do,
take a look at our <a href="getting_started/">Getting Started</a> page for instructions on how to start using Redmine and a
brief overview of what Redmine can do.</p>
<p>For any issues or general help with Redmine, see the <a href="troubleshooting/">Troubleshooting</a> page.</p>
<p><br></p>
<h2 id="possible-analyses">Possible Analyses</h2>
<h3 id="i-want-to">I want to:</h3>
<details>
  <summary><b>Download sequence data, or share sequence data with a collaborator</b></summary>  
  To retrieve a zip file of your sequence data (which can also be shared with external collaborators) use automators:  

   * [**External retrieve**](data/external_retrieve.md) - exports a list of sequences in a zipped file

</details>
<p><br></p>
<details>
  <summary><b>Download COWBAT reports for a list of sequence assemblies</b></summary>  
  To retrieve a zip file of the COWBAT assembly pipeline reports for your list of SeqIDs:  

   * [**Report retrieve**](data/report_retrieve.md) - exports a csv file of the legacy_combinedMetadata for your sequences, as well as a zip file containing all reports for those sequences.

</details>
<p><br></p>
<details>
  <summary><b>Detect gene(s) in my sequences</b></summary>  
  Automators that allow you to screen sequence(s) for gene targets:  

   * [**GeneSeekr**](analysis/geneseekr.md) - assemblies only  
   * [**KMA**](analysis/kma.md) - allows you to screen raw reads and assemblies
   * [**CARDRGI**](analysis/cardrgi.md) - *specific to AMR detection*

</details>
<p><br></p>
<details>
  <summary><b>Inspect the contigs of my Hybrid Assembly</b></summary>  
  Automators that allow you to retrieve gfa (graph files) for your hybrid assemblies:  

   * [**gfaretrieve**](analysis/gfa_retrieve.md) - hybrid assemblies only (MIN sequences)  


</details>
<p><br></p>
<details>
  <summary><b>Determine the genus/species of a whole genome sequence assembly</b></summary>  
  If you are unsure of the genus/species of your isolate, you can use the automator:  

   * [**Unknownisolate**](analysis/unknownisolate.md) - compares WGS assembly to ATCC and RefSeq genomes, and determines rMLST type. Outputs a GROBI report for probably identity.

</details>
<p><br></p>
<details>
  <summary><b>Find the <i>X</i> most diverse sequences in a set of sequences</b></summary>  
  If you want to find a number of diverse (e.g. distantly related) sequences from a set of sequences :  

   * [**diversitree**](analysis/diversitree.md) - will output a list based on user's requested number of sequences.

</details>
<p><br></p>
<details>
  <summary><b>Find sequences closely related to a query sequence</b></summary>  
  If you want to find <i>X</i> number of closest strains in a list to a query SeqID:  

   * [**NearTree**](analysis/neartree.md) - calculates the most closely related strains to your query SeqID based on MASH distance

</details>
<p><br></p>
<details>
  <summary><b>Determine taxonomy in metagenomes</b></summary>  
  If you want to detect the different organisms in your metagenomic sequence, you can use the automators:  

   * [**Metaphlan**](analysis/metaphlan.md) - more specific than Kraken2, but less sensitive.
   * [**Kraken2/Bracken**](analysis/kraken2.md) - more sensitive than metaphlan, but likely to give false positives to closely related genera/species. Bracken is more accurate than Kraken2 and Metaphlan4 (according to our publication [Cooper et al, 2023](https://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-023-03148-6)).

</details>
<p><br></p>
<details>
  <summary><b>Find SNPs in my sequences</b></summary>  
  To detect single nucleotide polymorphisms (SNPs) in your sequences, you can use the automators:  

   * [**SNVPhyl**](analysis/snvphyl.md) - 
   * [**Snippy**](analysis/snippy.md) - 

</details>
<p><br></p>
<details>
  <summary><b>Annotate my sequence assemblies</b></summary>  
  To annotate your sequence assemblies, you can use the automators:  

   * [**Prokka**](analysis/prokka.md) - whole genome annotation to identify features in gDNA (bacterial, archaeal, and viral)
   * **Bakta** - automator currently under development

</details>
<p><br></p>
<details>
  <summary><b>Create a phylogeny</b></summary>  
  To create a phylogenetic tree from a list of sequences, you can use the automators:  

   * [**MASHtree**](analysis/mashtree.md) - creates a phylogeny using MASH distances
   * [**bcgtree**](analysis/bcgtree.md) - builds a phylogeny using bacterial core genes ("107 essential single-copy core genes")
   * **iqtree** - automator currently under development

</details>
<p><br></p>
<details>
  <summary><b>Run a genome wide association study (GWAS)</b></summary>  
  You can use the automators:  

   * [**Roary/Scoary**](analysis/roary.md) - 
   * [**Pyseer**](analysis/pyseer.md) - 

</details>
<p><br></p>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="getting_started/" class="btn btn-neutral float-right" title="Getting Started">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/OLC-LOC-Bioinformatics/OLCRedmineAutomator" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
    
      <span><a href="getting_started/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script src="js/jquery-3.6.0.min.js"></script>
    <script>var base_url = ".";</script>
    <script src="js/theme_extra.js"></script>
    <script src="js/theme.js"></script>
      <script src="search/main.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>

<!--
MkDocs version : 1.6.0
Build Date UTC : 2024-07-10 19:36:01.620869+00:00
-->
