<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        <link rel="canonical" href="https://olc-bioinformatics.github.io/redmine-docs/analysis/roary/">
        <link rel="shortcut icon" href="../../img/favicon.ico">

	<title>Roary - OLC Redmine Automator</title>

        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../css/highlight.css">
        <link href="../../css/base.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../..">OLC Redmine Automator</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../..">Home</a>
                </li>
            
            
            
                <li >
                    <a href="../../getting_started/">Getting started</a>
                </li>
            
            
            
                <li >
                    <a href="../../troubleshooting/">Troubleshooting</a>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Tutorials <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../../tutorials/create_environment/">Create environment</a>
                        </li>
                    
                        <li >
                            <a href="../../tutorials/create_pages/">Create pages</a>
                        </li>
                    
                        <li >
                            <a href="../../tutorials/edit_pages/">Edit pages</a>
                        </li>
                    
                        <li >
                            <a href="../../tutorials/publish_changes/">Publish changes</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Data <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../../data/external_retrieve/">External retrieve</a>
                        </li>
                    
                        <li >
                            <a href="../../data/report_retrieve/">Report retrieve</a>
                        </li>
                    
                        <li >
                            <a href="../../data/sra_upload/">Sra upload</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Internal only <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../../internal_only/merge/">Merge</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Analysis <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../amrsummary/">Amrsummary</a>
                        </li>
                    
                        <li >
                            <a href="../autoclark/">Autoclark</a>
                        </li>
                    
                        <li >
                            <a href="../closerelatives/">Closerelatives</a>
                        </li>
                    
                        <li >
                            <a href="../confindr/">Confindr</a>
                        </li>
                    
                        <li >
                            <a href="../cowsnphr/">Cowsnphr</a>
                        </li>
                    
                        <li >
                            <a href="../diversitree/">Diversitree</a>
                        </li>
                    
                        <li >
                            <a href="../eCGF/">eCGF</a>
                        </li>
                    
                        <li >
                            <a href="../ectyper/">Ectyper</a>
                        </li>
                    
                        <li >
                            <a href="../geneseekr/">Geneseekr</a>
                        </li>
                    
                        <li >
                            <a href="../intimintyper/">Intimintyper</a>
                        </li>
                    
                        <li >
                            <a href="../mobsuite/">Mobsuite</a>
                        </li>
                    
                        <li >
                            <a href="../neartree/">Neartree</a>
                        </li>
                    
                        <li >
                            <a href="../plasmidborneidentity/">Plasmidborneidentity</a>
                        </li>
                    
                        <li >
                            <a href="../pointfinder/">Pointfinder</a>
                        </li>
                    
                        <li >
                            <a href="../primerfinder/">Primerfinder</a>
                        </li>
                    
                        <li >
                            <a href="../prokka/">Prokka</a>
                        </li>
                    
                        <li >
                            <a href="../psortb/">Psortb</a>
                        </li>
                    
                        <li >
                            <a href="../resfinder/">Resfinder</a>
                        </li>
                    
                        <li class="active">
                            <a href="./">Roary</a>
                        </li>
                    
                        <li >
                            <a href="../sipprverse/">Sipprverse</a>
                        </li>
                    
                        <li >
                            <a href="../snvphyl/">Snvphyl</a>
                        </li>
                    
                        <li >
                            <a href="../staramr/">Staramr</a>
                        </li>
                    
                        <li >
                            <a href="../strainmash/">Strainmash</a>
                        </li>
                    
                        <li >
                            <a href="../wgsassembly/">Wgsassembly</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li >
                    <a rel="next" href="../resfinder/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../sipprverse/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#roaryscoary">RoaryScoary</a></li>
        
            <li><a href="#what-does-it-do">What does it do?</a></li>
        
            <li><a href="#how-do-i-use-it">How do I use it?</a></li>
        
            <li><a href="#how-long-does-it-take">How long does it take?</a></li>
        
            <li><a href="#what-can-go-wrong">What can go wrong?</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<h1 id="roaryscoary">RoaryScoary</h1>
<h3 id="what-does-it-do">What does it do?</h3>
<h4 id="roary">Roary</h4>
<p>Roary is a pipeline for calculating pan genomes. Roary is not intended for metagenomics or for comparing extremely diverse sets of genomes. If you want to 
learn more about it, check out the <a href="https://academic.oup.com/bioinformatics/article/31/22/3691/240757">Roary publication</a>.</p>
<h4 id="scoary">Scoary</h4>
<p>Scoary takes the <code>gene_presence_absence.csv</code> file output from Roary and a traits file created by the user, and calculates the associations between the given traits and all genes in the accessory genome. For more information, check out the <a href="https://github.com/AdmiralenOla/Scoary">Scoary github</a> or <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-1108-8">Scoary publication</a>.</p>
<h3 id="how-do-i-use-it">How do I use it?</h3>
<h4 id="subject">Subject</h4>
<p>In the <code>Subject</code> field, put <code>roary</code>. Spelling counts, but case sensitivity doesn't.</p>
<h4 id="description">Description</h4>
<p><strong>Required Components</strong></p>
<p>In the <code>Description</code> field, you must provide the requested analysis type as follows:</p>
<p><code>analysistype=requested_analysis</code></p>
<p>The Roary automator supports the following analyses which perform operations on the pan genome of isolate sequences (again, spelling counts, but case sensitivity doesn't):</p>
<ul>
<li>union - reports the union of genes found in sequences</li>
<li>intersection - reports the intersection of genes found in isolate sequences (core genes)</li>
<li>complement - reports the complement of genes found in isolates (accessory genes)</li>
<li>gene_multifasta - extracts the sequence of each gene listed and creates multi-fasta files for each gene listed (outputs protein multi-fastas). <strong>NOTE</strong>: you must provide an additional line: <code>genes=gene1,gene2,geneN</code> which is a comma-separated list of genes you would like multi-fasta files for (eg. <code>genes=fliC,gyrA</code>). <strong>Case sensitivity counts for gene names</strong></li>
<li>difference - reports the gene differences between sets of isolates</li>
</ul>
<p>You must also include a list of SEQIDs one per line.</p>
<p><strong>Optional Components - Scoary</strong></p>
<p>If you would like scoary analysis to be completed, calculating the associations between traits and all genes in the accessory genome <strong>you must attach a csv file of traits (traits.csv) to the issue</strong>. Attaching a <code>traits.csv</code> file will automatically result in scoary analysis being conducted.</p>
<p><strong>This traits file must be formatted in a specific way</strong>.</p>
<ul>
<li>the rows should correspond to your isolate sequence IDs</li>
<li>the top right cell should be left blank</li>
<li>the trait(s) data must be binary (0 for trait absent, 1 for trait present)</li>
<li>all SEQ-IDs and traits should be uniquely named and not contain any weird characters (e.g. %;,/&amp;[]@? etc) </li>
</ul>
<p>The below table is an example from the Scoary github:</p>
<table>
<thead>
<tr>
<th align="center"></th>
<th align="center">Trait 1</th>
<th align="center">Trait 2</th>
<th align="center">...</th>
<th align="center">Trait N</th>
</tr>
</thead>
<tbody>
<tr>
<td align="center">YYYY-SEQ-0001</td>
<td align="center">1</td>
<td align="center">0</td>
<td align="center">...</td>
<td align="center">1</td>
</tr>
<tr>
<td align="center">YYYY-SEQ-0002</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">...</td>
<td align="center">1</td>
</tr>
<tr>
<td align="center">YYYY-SEQ-0003</td>
<td align="center">1</td>
<td align="center">0</td>
<td align="center">...</td>
<td align="center">1</td>
</tr>
</tbody>
</table>
<p>For more information, check out the <a href="https://github.com/AdmiralenOla/Scoary">Scoary github</a></p>
<h4 id="example">Example</h4>
<p>For an example Roary/Scoary issue see <a href="https://redmine.biodiversity.agr.gc.ca/issues/24968">issue 24968</a>, for an example Roary <code>analysistype=gene_multifasta</code> issue see <a href="https://redmine.biodiversity.agr.gc.ca/issues/25013">issue 25013</a>. (<strong>NOTE</strong>: the output files for these issues no longer exist on the ftp server.)</p>
<h4 id="interpreting-results">Interpreting Results</h4>
<p>The Roary automator will upload links to the ftp for files called <code>prokka_output.zip</code> and <code>roary_output.zip</code> once it has completed. </p>
<p>The <code>prokka_output.zip</code> contains all of the outputs from prokka. Prokka will output a lot of files for each genome you give it - you can find a quick description of
each file <a href="https://github.com/tseemann/prokka#output-files">here</a>. Of particular interest are the <code>.gff</code> files, which Roary uses for analysis.</p>
<p>The <code>roary_output.zip</code> file contains all of the outputs from roary. Some additional files will be present depending on the analysis type chosen:</p>
<ul>
<li>gene_multifasta - the zip file will include <code>.fa</code> files that contain the aligned protein sequences for the requested gene(s) (one for file each gene included in the analysis, eg. <code>gene_multifasta_fliC.fa</code> and <code>gene_multifasta_gyrA.fa</code>). If you'd like, these sequences can be further analysed using NCBI protein blast.</li>
<li>scoary - the zip file will include <code>.csv</code> file(s) for each of the traits (column headers) provided in the attached <code>traits.csv</code> file.</li>
</ul>
<h3 id="how-long-does-it-take">How long does it take?</h3>
<p>Prokka isn't the quickest thing around - expect it to take 2 to 3 minutes for each genome you give it. After prokka is finished the Roary pipeline time will depend on the analysis type, and the number of genes/traits requested (in the case of gene_multifasta and scoary). </p>
<h3 id="what-can-go-wrong">What can go wrong?</h3>
<ol>
<li>Requested SEQIDs are not available. If we can't find some of the SEQIDs that you request, you will get a warning message informing you of it.</li>
<li>There was an issue with the requested analysistype: either one was not supplied, the was a typo, or you requested a currently-unsupported analysis. An error message detailing the problem will be added to the issue.</li>
<li><strong>Scoary</strong> You will not get an error if the traits.csv file is not attached or formatted correctly. Roary will complete, but no <code>.csv</code> files for each of the traits will be included in the zip file. Unfortunately, you will have to submit the analysis request again with the correct <code>traits.csv</code> formatting.</li>
</ol></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../js/jquery-1.10.2.min.js"></script>
        <script src="../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../js/highlight.pack.js"></script>
        <script src="../../js/base.js"></script>
    </body>
</html>