<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Snippy - OLC Redmine Automator</title>
  

  <link rel="shortcut icon" href="../../img/favicon.ico">

  
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="../../css/highlight.css">

  
  <script>
    // Current page data
    var mkdocs_page_name = "Snippy";
  </script>
  
  <script src="../../js/jquery-2.1.1.min.js"></script>
  <script src="../../js/modernizr-2.8.3.min.js"></script>
  <script type="text/javascript" src="../../js/highlight.pack.js"></script>
  <script src="../../js/theme.js"></script> 

  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href="../.." class="icon icon-home"> OLC Redmine Automator</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
        <ul class="current">
          
            <li>
    <li class="toctree-l1 ">
        <a class="" href="../..">Home</a>
        
    </li>
<li>
          
            <li>
    <li class="toctree-l1 ">
        <a class="" href="../../getting_started/">Getting started</a>
        
    </li>
<li>
          
            <li>
    <li class="toctree-l1 ">
        <a class="" href="../../troubleshooting/">Troubleshooting</a>
        
    </li>
<li>
          
            <li>
    <ul class="subnav">
    <li><span>Analysis</span></li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../amrsummary/">Amrsummary</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../autoclark/">Autoclark</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../bcgtree/">Bcgtree</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../cardrgi/">Cardrgi</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../closerelatives/">Closerelatives</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../confindr/">Confindr</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../cowsnphr/">Cowsnphr</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../diversitree/">Diversitree</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../drep/">Drep</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../eCGF/">eCGF</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../ectyper/">Ectyper</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../fastqc/">Fastqc</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../geneseekr/">Geneseekr</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../intimintyper/">Intimintyper</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../kma/">Kma</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../kraken2/">Kraken2</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../mashtree/">Mashtree</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../metaphlan/">Metaphlan</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../mobsuite/">Mobsuite</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../neartree/">Neartree</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../plasmidborneidentity/">Plasmidborneidentity</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../pointfinder/">Pointfinder</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../primerfinder/">Primerfinder</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../prokka/">Prokka</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../psortb/">Psortb</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../pyseer/">Pyseer</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../resfinder/">Resfinder</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../roary/">Roary</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../sequence_extractor/">Sequence extractor</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../sipprverse/">Sipprverse</a>
        
    </li>

        
            
    <li class="toctree-l1 current">
        <a class="current" href="./">Snippy</a>
        
            <ul>
            
                <li class="toctree-l3"><a href="#snippy">Snippy</a></li>
                
                    <li><a class="toctree-l4" href="#what-does-it-do">What does it do?</a></li>
                
                    <li><a class="toctree-l4" href="#how-do-i-use-it">How do I use it?</a></li>
                
                    <li><a class="toctree-l4" href="#variant-types">Variant types</a></li>
                
                    <li><a class="toctree-l4" href="#how-long-does-it-take">How long does it take?</a></li>
                
                    <li><a class="toctree-l4" href="#what-can-go-wrong">What can go wrong?</a></li>
                
                    <li><a class="toctree-l4" href="#should-i-use-snippy-or-snvphyl">Should I use Snippy or SNVPhyl?</a></li>
                
            
            </ul>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../snvphyl/">Snvphyl</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../staramr/">Staramr</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../strainmash/">Strainmash</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../wgsassembly/">Wgsassembly</a>
        
    </li>

        
    </ul>
<li>
          
            <li>
    <ul class="subnav">
    <li><span>Data</span></li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../../data/external_retrieve/">External retrieve</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../../data/report_retrieve/">Report retrieve</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../../data/sra_upload/">Sra upload</a>
        
    </li>

        
    </ul>
<li>
          
            <li>
    <ul class="subnav">
    <li><span>Tutorials</span></li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../../tutorials/create_environment/">Create environment</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../../tutorials/create_pages/">Create pages</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../../tutorials/edit_pages/">Edit pages</a>
        
    </li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../../tutorials/publish_changes/">Publish changes</a>
        
    </li>

        
    </ul>
<li>
          
            <li>
    <ul class="subnav">
    <li><span>Internal only</span></li>

        
            
    <li class="toctree-l1 ">
        <a class="" href="../../internal_only/merge/">Merge</a>
        
    </li>

        
    </ul>
<li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../..">OLC Redmine Automator</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../..">Docs</a> &raquo;</li>
    
      
        
          <li>Analysis &raquo;</li>
        
      
    
    <li>Snippy</li>
    <li class="wy-breadcrumbs-aside">
      
        
          <a href="https://github.com/OLC-LOC-Bioinformatics/OLCRedmineAutomator" class="icon icon-github"> Edit on GitHub</a>
        
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="snippy">Snippy</h1>
<h3 id="what-does-it-do">What does it do?</h3>
<p>Snippy is a tool for rapid haploid variant calling and core genome alignment of closely related genomes. This automator can also be used to build a phylogenetic tree to attempt to show the
relatedness of these strains. Lots more info can be found at the <a href="https://github.com/tseemann/snippy">Snippy github site</a>.</p>
<h3 id="how-do-i-use-it">How do I use it?</h3>
<h4 id="subject">Subject</h4>
<p>In the <code>Subject</code> field, put <code>snippy</code>. Spelling counts, but case sensitivity doesn't.</p>
<h4 id="description">Description</h4>
<p><strong>Required Components</strong></p>
<p>The first line of your description needs to be <code>reference=</code>, the SEQID of the strain you want to act
as your reference strain. Ideally, you'll want to pick a high-quality assembly for your reference.</p>
<p>eg. <code>reference=YYYY-MIN-NNNN</code></p>
<p>If you wish to attach a reference file instead of providing a SEQID, the line must be <code>reference=attached</code></p>
<p>You must also include a list of SEQIDs one per line.</p>
<p><strong>Optional Components</strong></p>
<p>We have also included the ability to 'cleanup' a multiple sequence alignment, and generate a tree using <a href="http://nickjcroucher.github.io/gubbins/">Gubbins</a> and <a href="http://www.iqtree.org/">IQ-tree</a>. If you call SNPs for multiple isolates from the same reference, you are able to produce an alignment of "core SNPs" to build a phylogeny. To use this function, add the following line in the description:</p>
<ul>
<li><code>cleanup_&amp;_tree=True</code></li>
</ul>
<p>This automator currently uses IQtree and default settings for gubbins, but other options are available. If you would like additional functions, please contact a bioinformatician and we will do our best to add them.</p>
<h4 id="example">Example</h4>
<p>For an example snippy, see <a href="https://redmine.biodiversity.agr.gc.ca/issues/30388">issue 30388</a>.</p>
<h4 id="interpreting-results">Interpreting Results</h4>
<p>The zip file uploaded on snippy completion will contain folders for each of the comparator sequences, as well as core alignment files. Some of the files will also be uploaded/attached directly to the redmine request. Important files are:</p>
<ul>
<li><code>core.txt</code>: A summary file. Tab-separated columnar list of alignment/core-size statistics of the number of: aligned, unaligned, variants, and low coverage bases between every strain submitted and the reference.</li>
<li><code>core.tab</code>: Tab-separated columnar list of core SNP sites with alleles but NO annotations</li>
<li><code>core.vcf</code>: Multi-sample VCF file with genotype GT tags for all discovered alleles</li>
<li>Within each sequence folder there will also be files, including a tab separated summary for that particular sequence in comparison to the reference.</li>
</ul>
<p>If cleanup_&amp;_tree was included, then tree files will also be included in the output:</p>
<ul>
<li><code>{issue.id}.snippy.gubbins.iqtree.tree</code>: An intermediate phylogenetic tree created by gubbins + iqtree. </li>
<li><code>{issue.id}.snippy_final.clean.core.snp.aln.tree</code>: The <strong>final</strong> phylogeny created from the core snp alignment (created using IQtree and the GTR substitution method).</li>
</ul>
<p>If you want to view these tree files, you can use a program such as
<a href="http://tree.bio.ed.ac.uk/software/figtree/">FigTree</a> or a web-based viewer like <a href="http://phylo.io">phylo.io</a>.</p>
<p>Other files can also be important - see the <a href="https://github.com/tseemann/snippy">docs on Snippy Output files</a>
for more information.</p>
<h3 id="variant-types">Variant types</h3>
<p>For more information, see the <a href="https://github.com/tseemann/snippy">docs on Snippy Output files</a>:</p>
<table>
<thead>
<tr>
<th align="center">Type</th>
<th align="center">Name</th>
<th align="center">Example</th>
</tr>
</thead>
<tbody>
<tr>
<td align="center">snp</td>
<td align="center">Single Nucleotide Polymorphism</td>
<td align="center">A =&gt; T</td>
</tr>
<tr>
<td align="center">mnp</td>
<td align="center">Multiple Nuclotide Polymorphism</td>
<td align="center">GC =&gt; AT</td>
</tr>
<tr>
<td align="center">ins</td>
<td align="center">Insertion</td>
<td align="center">ATT =&gt; AGTT</td>
</tr>
<tr>
<td align="center">del</td>
<td align="center">Deletion</td>
<td align="center">ACGG =&gt; ACG</td>
</tr>
<tr>
<td align="center">complex</td>
<td align="center">Combination of snp/mnp</td>
<td align="center">ATTC =&gt; GTTA</td>
</tr>
</tbody>
</table>
<h3 id="how-long-does-it-take">How long does it take?</h3>
<p>It depends on the number of sequences and coverage/size of sequence files. Small snippy requests take ~15 minutes to complete (see the example request). If you submit a request for a larger snippy (&gt;30 strains), or include larger long-read sequences, it may take substantially longer.</p>
<h3 id="what-can-go-wrong">What can go wrong?</h3>
<p>A few things can go wrong with this process:</p>
<p>1) Requested SEQIDs are not available. If we can't find some of the SEQIDs that you request, you will get a warning
message informing you of it.</p>
<p>2) Strains too far apart. Snippy requires that the strains you want to compare to the reference be closely related to
the reference. If you ask for a snippy-analysis with things that are not very related, the analysis will fail. In this case, look at the <code>core.txt</code> file that is uploaded to the redmine request. If there are sequences with 0 under "Aligned", remove those sequences from the request and re-submit.</p>
<p>3) Why is the cleaned snp tree not showing my single-end sequences? - I'm not sure... the cleaning step(s) using gubbins appears to remove the minion (single-end) sequence data from the cleaned alignment when a multi-analysis is performed using both illumina and minion sequences. The <code>{}.snippy_core_alignment.iqtree.treefile</code>, which is generated before the cleaning steps, will contain all SEQIDS.</p>
<h3 id="should-i-use-snippy-or-snvphyl">Should I use Snippy or SNVPhyl?</h3>
<p>...This depends on your goal...</p>
<p>Snippy appears to be faster than SNVPhyl. It also allows use of both paired-end and single-end raw data files in the same analysis (whereas SNVPhyl <strong>currently does not</strong>).</p>
<p>A comparison of snippy vs SNVPhyl has not been conducted by our lab. The <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5628696/">SNVPhyl publication</a> does briefly discuss snippy. </p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../snvphyl/" class="btn btn-neutral float-right" title="Snvphyl"/>Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../sipprverse/" class="btn btn-neutral" title="Sipprverse"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
	  
        </div>
      </div>

    </section>

  </div>

<div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
          <a class="icon icon-github" style="float: left; color: #fcfcfc"> GitHub</a>
      
      
        <span><a href="../sipprverse/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../snvphyl/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>

</body>
</html>
