<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        <link rel="canonical" href="https://olc-bioinformatics.github.io/redmine-docs/analysis/cowsnphr/">
        <link rel="shortcut icon" href="../../img/favicon.ico">

	<title>COWSNPhR - OLC Redmine Automator</title>

        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../css/highlight.css">
        <link href="../../css/base.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../..">OLC Redmine Automator</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../..">Home</a>
                </li>
            
            
            
                <li >
                    <a href="../../getting_started/">Getting Started</a>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Data <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../../data/external_retrieve/">External Retrieve</a>
                        </li>
                    
                        <li >
                            <a href="../../data/report_retrieve/">Report Retrieve</a>
                        </li>
                    
                        <li >
                            <a href="../../data/sra_upload/">SRA Upload</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Analysis <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../amrsummary/">AMRsummary</a>
                        </li>
                    
                        <li >
                            <a href="../autoclark/">AutoCLARK</a>
                        </li>
                    
                        <li >
                            <a href="../confindr/">ConFindr</a>
                        </li>
                    
                        <li class="active">
                            <a href="./">COWSNPhR</a>
                        </li>
                    
                        <li >
                            <a href="../closerelatives/">CloseRelatives</a>
                        </li>
                    
                        <li >
                            <a href="../diversitree/">DiversiTree</a>
                        </li>
                    
                        <li >
                            <a href="../geneseekr/">GeneSeekr</a>
                        </li>
                    
                        <li >
                            <a href="../mobsuite/">MobSuite</a>
                        </li>
                    
                        <li >
                            <a href="../neartree/">NearTree</a>
                        </li>
                    
                        <li >
                            <a href="../plasmidborneidentity/">PlasmidBorneIdentity</a>
                        </li>
                    
                        <li >
                            <a href="../pointfinder/">PointFinder</a>
                        </li>
                    
                        <li >
                            <a href="../prokka/">Prokka</a>
                        </li>
                    
                        <li >
                            <a href="../resfinder/">ResFinder</a>
                        </li>
                    
                        <li >
                            <a href="../sipprverse/">sipprverse</a>
                        </li>
                    
                        <li >
                            <a href="../snvphyl/">SNVPhyl</a>
                        </li>
                    
                        <li >
                            <a href="../staramr/">staramr</a>
                        </li>
                    
                        <li >
                            <a href="../strainmash/">StrainMash</a>
                        </li>
                    
                        <li >
                            <a href="../wgsassembly/">WGS Assembly</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Internal Use <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../../internal_only/merge/">Merge</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            
                <li >
                    <a href="../../troubleshooting/">Troubleshooting</a>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li >
                    <a rel="next" href="../confindr/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../closerelatives/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#cowsnphr">COWSNPhR</a></li>
        
            <li><a href="#what-does-it-do">What does it do?</a></li>
        
            <li><a href="#how-do-i-use-it">How do I use it?</a></li>
        
            <li><a href="#how-long-does-it-take">How long does it take?</a></li>
        
            <li><a href="#what-can-go-wrong">What can go wrong?</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<h1 id="cowsnphr">COWSNPhR</h1>
<h3 id="what-does-it-do">What does it do?</h3>
<p>COWSNPhR (CFIA OLC Workflow for Single Nucleotide PHylogeny Reporting) is a pipeline developed in conjunction with the 
USDA APHIS Veterinary Services that determines the presence of high quality Single Nucleotide Variants between a 
reference strain and other closely related strains. Following annotation of the reference strain, using 
<a href="https://github.com/tseemann/prokka">Prokka</a>, the pipeline maps the raw reads of the query strains to the reference 
genome using <a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">bowtie2</a>, finds genetic variants with 
<a href="https://github.com/google/deepvariant">DeepVariant</a>, extracts variants, creates multiple sequence alignments, and
calculates a phylogenetic tree using <a href="http://www.microbesonline.org/fasttree/">FastTree</a> to show the relatedness of 
these strains. The location of each variant is mapped to the annotation of the reference strain, and all variants are 
entered in a summary table.</p>
<h3 id="how-do-i-use-it">How do I use it?</h3>
<h4 id="subject">Subject</h4>
<p>In the <code>Subject</code> field, put <code>cowsnphr</code>. Spelling counts, but case sensitivity does not.</p>
<h4 id="description">Description</h4>
<p>The first line of your description needs to be <code>reference</code>, and the second line the SEQID of the strain you want to act
as your reference strain. Ideally, you'll want to pick a high-quality assembly for your reference.</p>
<p>If you wish to attach a reference file instead of providing a SEQID, the second line must be <code>attached</code></p>
<p>The third line of your description should be <code>compare</code>, and lines after that the SEQIDs for strains to which you want to 
compare your reference.</p>
<h4 id="example">Example</h4>
<p>For example COWSNPhR analyses, see <a href="https://redmine.biodiversity.agr.gc.ca/issues/15681">issue 15681</a> and 
<a href="https://redmine.biodiversity.agr.gc.ca/issues/15682">issue 15682 (uploaded reference file)</a>.</p>
<h4 id="interpreting-results">Interpreting Results</h4>
<p>The zip file uploaded on COWSNPhR completion should contain four folders: </p>
<ul>
<li><code>vcf_files</code>: compressed global VCF files </li>
<li><code>summary_tables</code>: table summarizing the location, prevalence, and annotation of variants</li>
<li><code>alignments</code>: multiple sequence alignment of variants</li>
<li><code>tree_files</code>: phylogenetic tree of alignment. If you want to view this tree, you can use a program such as
<a href="http://tree.bio.ed.ac.uk/software/figtree/">FigTree</a> or a web-based viewer like <a href="http://phylo.io">phylo.io</a>.</li>
</ul>
<h3 id="how-long-does-it-take">How long does it take?</h3>
<p>Most COWSNPhR requests take ~1 hour to complete. If you submit a request for a larger COWSNPhR analysis (&gt;30 strains), 
it may take substantially longer.</p>
<h3 id="what-can-go-wrong">What can go wrong?</h3>
<p>A few things can go wrong with this process:</p>
<p>1) Requested SEQIDs are not available. If we can't find some of the SEQIDs that you request, you will get a warning
message informing you of it.</p>
<p>2) Strains too far apart. COWSNPhR requires that the strains you want to compare to the reference be closely related to
the reference. If you ask for an analysis with strains that are not very related, you will get a warning telling you so.</p>
<p>3) Other errors. This software is still in active development, so there may be unforeseen issues arising for novel 
reference: query combinations. </p></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../js/jquery-1.10.2.min.js"></script>
        <script src="../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../js/highlight.pack.js"></script>
        <script src="../../js/base.js"></script>
    </body>
</html>