<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        <link rel="canonical" href="https://olc-bioinformatics.github.io/redmine-docs/analysis/plasmidborneidentity/">
        <link rel="shortcut icon" href="../../img/favicon.ico">

	<title>PlasmidBorneIdentity - OLC Redmine Automator</title>

        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../css/highlight.css">
        <link href="../../css/base.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../..">OLC Redmine Automator</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../..">Home</a>
                </li>
            
            
            
                <li >
                    <a href="../../getting_started/">Getting Started</a>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Data <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../../data/external_retrieve/">External Retrieve</a>
                        </li>
                    
                        <li >
                            <a href="../../data/report_retrieve/">Report Retrieve</a>
                        </li>
                    
                        <li >
                            <a href="../../data/sra_upload/">SRA Upload</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Analysis <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../amrsummary/">AMRsummary</a>
                        </li>
                    
                        <li >
                            <a href="../autoclark/">AutoCLARK</a>
                        </li>
                    
                        <li >
                            <a href="../confindr/">ConFindr</a>
                        </li>
                    
                        <li >
                            <a href="../cowsnphr/">COWSNPhR</a>
                        </li>
                    
                        <li >
                            <a href="../closerelatives/">CloseRelatives</a>
                        </li>
                    
                        <li >
                            <a href="../ectyper/">ECTyper</a>
                        </li>
                    
                        <li >
                            <a href="../diversitree/">DiversiTree</a>
                        </li>
                    
                        <li >
                            <a href="../geneseekr/">GeneSeekr</a>
                        </li>
                    
                        <li >
                            <a href="../mobsuite/">MobSuite</a>
                        </li>
                    
                        <li >
                            <a href="../neartree/">NearTree</a>
                        </li>
                    
                        <li class="active">
                            <a href="./">PlasmidBorneIdentity</a>
                        </li>
                    
                        <li >
                            <a href="../pointfinder/">PointFinder</a>
                        </li>
                    
                        <li >
                            <a href="../primerfinder/">PrimerFinder</a>
                        </li>
                    
                        <li >
                            <a href="../prokka/">Prokka</a>
                        </li>
                    
                        <li >
                            <a href="../psortb/">PSORTb</a>
                        </li>
                    
                        <li >
                            <a href="../resfinder/">ResFinder</a>
                        </li>
                    
                        <li >
                            <a href="../sipprverse/">sipprverse</a>
                        </li>
                    
                        <li >
                            <a href="../snvphyl/">SNVPhyl</a>
                        </li>
                    
                        <li >
                            <a href="../staramr/">staramr</a>
                        </li>
                    
                        <li >
                            <a href="../strainmash/">StrainMash</a>
                        </li>
                    
                        <li >
                            <a href="../wgsassembly/">WGS Assembly</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Internal Use <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../../internal_only/merge/">Merge</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            
                <li >
                    <a href="../../troubleshooting/">Troubleshooting</a>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li >
                    <a rel="next" href="../neartree/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../pointfinder/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#plasmid-borne-identity">Plasmid-Borne Identity</a></li>
        
            <li><a href="#what-does-it-do">What does it do?</a></li>
        
            <li><a href="#how-do-i-use-it">How do I use it?</a></li>
        
            <li><a href="#how-long-does-it-take">How long does it take?</a></li>
        
            <li><a href="#what-can-go-wrong">What can go wrong?</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<h1 id="plasmid-borne-identity">Plasmid-Borne Identity</h1>
<p><img alt="Plasmid-Borne Identity" src="../../img/plasmid_borne_identity.png" /></p>
<h3 id="what-does-it-do">What does it do?</h3>
<p>Plasmid-Borne Identity performs GeneSeekr and MOB-suite analyses on FASTA files, and creates a report combining and 
summarising the outputs.</p>
<p>MobSuite is a set of tools developed by the Public Health Agency of Canada for detecting plasmids in draft genome
assemblies. This tool runs the <code>mob_recon</code> part of the suite, which first detects plasmids in the assemblies, and then
performs typing on the plasmids. More details on MobSuite, including fairly extensive details on the output files
produced, can be found at the <a href="https://github.com/phac-nml/mob-suite">MobSuite GitHub repository</a>).</p>
<h3 id="how-do-i-use-it">How do I use it?</h3>
<h4 id="subject">Subject</h4>
<p>In the <code>Subject</code> field, put <code>plasmid_borne_identity</code>. Spelling counts, but case sensitivity doesn't.</p>
<h4 id="description">Description</h4>
<p>All you need to put in the description is a list of SEQIDs you want to process, one per line.</p>
<h4 id="attachments">Attachments</h4>
<p>You are required to attach a FASTA-formatted file containing the gene(s) you wish analysed </p>
<h4 id="optional-arguments">Optional arguments</h4>
<ul>
<li>BLAST program. <strong>NOTE:</strong> GeneSeekr (and therefore PlasmidBorne Identity) does not check to see if your query or 
database are the appropriate molecule for the requested program. <ul>
<li>default is <code>blastn</code></li>
<li>You can select one of the following BLAST programs to use:<ul>
<li>blastn - nt query: nt db</li>
<li>blastp - protein query: protein db</li>
<li>blastx - translated nt query: protein db</li>
<li>tblastn - protein query: translated nt db</li>
<li>tblastx - translated nt query: translated nt db</li>
</ul>
</li>
<li>modify as follows:<ul>
<li><code>blast=tblastx</code></li>
</ul>
</li>
</ul>
</li>
<li>Minimum cutoff for matches to be included in report.<ul>
<li>default is <code>70</code></li>
<li>modify as follows:<ul>
<li><code>cutoff=80</code></li>
</ul>
</li>
</ul>
</li>
<li>E-value cutoff<ul>
<li>default is <code>1E-05</code></li>
<li>modify as follows:<ul>
<li><code>evalue=1E-10</code> or</li>
<li><code>evalue=0.01</code></li>
</ul>
</li>
</ul>
</li>
</ul>
<h4 id="example">Example</h4>
<p>For an example Plasmid-Borne Identity analysis, see <a href="https://redmine.biodiversity.agr.gc.ca/issues/15644">issue 15644</a>.</p>
<h4 id="interpreting-results">Interpreting Results</h4>
<p>Plasmid-Borne Identity will upload five separate reports once it is complete</p>
<p><code>geneseekr_{BLAST_PROGRAM}.xlsx</code> strain name and percent identity match for all query genes</p>
<p><code>geneseekr_{BLAST_PROGRAM}_detailed.csv</code> strain name, and BLAST summary information, including percent match, alignment length, 
subject length, e-value, number of positives, number of mismatches, and number of gaps for every gene</p>
<p><code>geneseekr_{BLAST_PROGRAM}.csv</code>  same as <code>geneseekr_{BLAST_PROGRAM}.csv</code>  same as, but in .csv format</p>
<p><code>mob_recon_summary.csv</code> shows any contigs that are predicted to be plasmids - note that all contigs calculated to be 
chromosomal are ignored. <strong>Location</strong> is the name of the predicted plasmid, while <strong>Contig</strong> is the name contig 
predicted to contain plasmid sequence. One plasmid can be composed of several contigs if it could not be circularised.</p>
<p><code>plasmid_borne_summary.csv</code> combines information from <code>geneseekr_{BLAST_PROGRAM}.csv</code>  and <code>mob_recon_summary.csv</code>. 
The contigs of all predicted AMR genes from the <code>geneseekr_{BLAST_PROGRAM}_detailed.csv</code> report are used to search 
the <code>mob_recon_summary</code> report. The plasmid predictions, and well as all the incompatibility types for that plasmid 
are extracted, and used in the report. <strong>Location</strong> will specify either <code>chromosome</code> or the name of the predicted plasmid.</p>
<h3 id="how-long-does-it-take">How long does it take?</h3>
<p>GeneSeekr is very fast, while MOB-suite is relatively slow - it should take a few minutes to analyze each SEQID requested.</p>
<h3 id="what-can-go-wrong">What can go wrong?</h3>
<ol>
<li>Requested SEQIDs are not available. If we can't find some of the SEQIDs that you request, you will get a warning
message informing you of it.</li>
<li>Issue with required FASTA-formatted targets file, including not attaching the file, or the FASTA formatting being incorrect</li>
<li>Specifying the incorrect BLAST analysis program for the provided sequences e.g. blastp with a nucleotide query and db</li>
</ol></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../js/jquery-1.10.2.min.js"></script>
        <script src="../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../js/highlight.pack.js"></script>
        <script src="../../js/base.js"></script>
    </body>
</html>